Assets/Scripts/Interface/Plan Wizard/PlanWizard.cs

```diff
@@ -317,7 +317,27 @@ private void SubmitEditPlanChanges(BatchRequest batch)
             if (energyLayersRemoved)
                 needsEnergyError = true;
 
-            bool typeChanged = false;
+			//If any layers are removed, issues need to be rechecked without these layers
+			if (layersToRemove != null && layersToRemove.Count > 0)
+			{
+				foreach (AbstractLayer layer in layersToRemove)
+				{
+					layer.RemovePlanLayer(editingPlan.GetPlanLayerForLayer(layer));
+				}
+
+				RestrictionIssueDeltaSet issuesToSubmit = ConstraintManager.CheckConstraints(editingPlan, IssueManager.instance.FindIssueDataForPlan(editingPlan), true, layersToRemove);
+				if (issuesToSubmit != null)
+				{
+					issuesToSubmit.SubmitToServer(batch);
+				}
+				//TODO: if submission fails, restore old issues
```

Comment:
Is this comment referring to: ?
`foreach (AbstractLayer layer in layersToRemove)
{
    layer.AddPlanLayer(editingPlan.GetPlanLayerForLayer(layer));
}`

In that case where do we check if the submission fails?
If that is done, then it should be better to remove TODO comments (but keep the explanation).

If instead this is not pertaining at all the current task/pr, then it's all good for me :)